metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -996,18 +1004,25 @@ def visit_constrain_time_range_node(self, node: ConstrainTimeRangeNode) -> SqlDa
             ds >= '2020-01-01' AND ds <= '2020-02-01'
 
         instead of this: DATE_TRUNC('month', ds) >= '2020-01-01' AND DATE_TRUNC('month', ds <= '2020-02-01')
+
+        Since time range constraints are always bound by a range of standard date/time values, this conversion
```

Comment:
Maybe we should allow custom granularities here? Won't they be date/time values, too? We could enforce that if needed, but maybe that's too restrictive.

Reply:
The values can be arbitrary, there's no guarantee they look like date/time at all. I can have a retail calendar with entries like Y2024Q01M02 and Y2024Q02M02.

For something like that we'd have to do the range adjustment based on a query against the time spine, which we can't do at query planning time, or by doing a full join. Either way I don't think it's worth the effort.

There is an interesting issue here - a custom grain defined in a cumulative metric means we'll never be able to do time range constraint predicate push down for that metric unless we get some added input from the user.

Reply:
Well, can't we apply a time range constraint on the base grain?